src/main/java/org/spongepowered/api/world/map/MapStorage.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world.map;
+
+import org.spongepowered.api.data.manipulator.mutable.item.MapItemData;
+
+import java.util.Optional;
+
+public interface MapStorage {
+    /**
+     * Gets the highest known map id if there are any
+     * May be inaccurate if maps overflow
+     * @return int Highest known map
+     */
+    Optional<Integer> getHighestMapId();
+
+    /**
+     * Gets {@link MapItemData}
+     * @param id int Map id
+     * @return MapItemData of that map id.
+     */
+    Optional<MapItemData> getMapData(int id);
```

Comment:
To make a point, I do understand that we should have a Map data representation, but I feel that we do somewhat need some registry of sorts, unfortunately, Maps use just an `int` id system, whereas the API doesn't really need to adhere to the id system to retain a map, I feel we could have a world-based id system in place for the world stored maps, versus the server-sided/virtual maps that plugins create and send to clients. Would this be doable?